# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ImageContainer.cpp
# Commit: 6149f5b15c5e
# Full Hash: 6149f5b15c5e278ef15ea36fa896fa8c146ebdca
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-14 08:32:45
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 29fd3a6a7e27 -r 6149f5b15c5e gfx/layers/ImageContainer.cpp
--- a/gfx/layers/ImageContainer.cpp	Wed Jan 13 20:14:54 2021 +0000
+++ b/gfx/layers/ImageContainer.cpp	Wed Jan 13 21:49:40 2021 +0000
@@ -225,6 +225,24 @@
   }
 }
 
+Maybe<SurfaceDescriptor> Image::GetDesc() { return {}; }
+
+Maybe<SurfaceDescriptor> Image::GetDescFromTexClient(
+    TextureClient* const forTc) {
+  RefPtr<TextureClient> tc = forTc;
+  if (!forTc) {
+    tc = GetTextureClient(nullptr);
+  }
+
+  const auto& tcd = tc->GetInternalData();
+
+  SurfaceDescriptor ret;
+  if (!tcd->Serialize(ret)) {
+    return {};
+  }
+  return Some(ret);
+}
+
 RefPtr<PlanarYCbCrImage> ImageContainer::CreatePlanarYCbCrImage() {
   RecursiveMutexAutoLock lock(mRecursiveMutex);
   EnsureImageClient();